cli/cli.c

```diff
@@ -0,0 +1,201 @@
+{
+    (void)handle;
+
+    buf->base = malloc(suggested_size);
+    buf->len = suggested_size;
+}
+
+static void shutdown_cb(uv_shutdown_t* req, int status)
+{
+    int ret;
+
+    (void)req;
+    (void)status;
+
+    /* receive reply */
+    response_string_size = 0;
+    response_string[0] = '\0';
+
+    ret = uv_read_start((uv_stream_t *)&client_pipe, alloc_cb, pipe_read_cb);
+    if (ret) {
+        fprintf(stderr, "uv_read_start(): %s\n", uv_strerror(ret));
+        uv_close((uv_handle_t *)&client_pipe, NULL);
+        return;
+    }
+
+}
+
+static void pipe_write_cb(uv_write_t* req, int status)
+{
+    int ret;
+
+    (void)req;
+    (void)status;
+
+    ret = uv_shutdown(&shutdown_req, (uv_stream_t *)&client_pipe, shutdown_cb);
+    if (ret) {
+        fprintf(stderr, "uv_shutdown(): %s\n", uv_strerror(ret));
+        uv_close((uv_handle_t *)&client_pipe, NULL);
+        return;
+    }
+}
+
+static void connect_cb(uv_connect_t* req, int status)
+{
+    int ret;
+    uv_buf_t write_buf;
+    char *s;
+
+    (void)req;
+    if (status) {
+        fprintf(stderr, "uv_pipe_connect(): %s\n", uv_strerror(status));
+        fprintf(stderr, "Make sure the netdata service is running.\n");
+        exit(-1);
+    }
+    if (0 == command_string_size) {
+        s = fgets(command_string, MAX_COMMAND_LENGTH, stdin);
+    }
+    (void)s; /* We don't need input to communicate with the server */
+    command_string_size = strlen(command_string);
+
+    write_req.data = &client_pipe;
+    write_buf.base = command_string;
+    write_buf.len = command_string_size;
+    ret = uv_write(&write_req, (uv_stream_t *)&client_pipe, &write_buf, 1, pipe_write_cb);
+    if (ret) {
+        fprintf(stderr, "uv_write(): %s\n", uv_strerror(ret));
+    }
+//  fprintf(stderr, "COMMAND: Sending command: \"%s\"\n", command_string);
+}
+
+int main(int argc, char **argv)
+{
+    int ret, i;
+    static uv_loop_t* loop;
+    uv_connect_t req;
+
+    exit_status = -1; /* default status for when there is no command response from server */
+
+    loop = uv_default_loop();
+
+    ret = uv_pipe_init(loop, &client_pipe, 1);
+    if (ret) {
+        fprintf(stderr, "uv_pipe_init(): %s\n", uv_strerror(ret));
+        return exit_status;
+    }
+
+    command_string_size = 0;
+    command_string[0] = '\0';
+    for (i = 1 ; i < argc ; ++i) {
+        size_t to_copy;
+
+        to_copy = MIN(strlen(argv[i]), MAX_COMMAND_LENGTH - 1 - command_string_size);
+        strncpyz(command_string + command_string_size, argv[i], to_copy);
+        command_string_size += to_copy;
+
+        if (command_string_size < MAX_COMMAND_LENGTH - 1) {
+            command_string[command_string_size++] = ' ';
+        } else {
+            break;
+        }
+    }
+
+    uv_pipe_connect(&req, &client_pipe, PIPENAME, connect_cb);
+
+    uv_run(loop, UV_RUN_DEFAULT);
+
+    uv_close((uv_handle_t *)&client_pipe, NULL);
+
+    return exit_status;
+}
```

Comment:
Missing new line here.

Reply:
This is not good tonight - please no syntax. This must be closed.